pages/v4/Coroutines.md

```diff
@@ -0,0 +1,184 @@
+
+To get back to our previous example, we could have our Coroutine as follows:
+
+```csharp
+private IEnumerable<WaitUntil> MoveTo(Vector2 target)
+{
+	Vector2 movement = (target - this.GameObj.Transform.Pos.Xy).Normalized;
+	
+	while((target - this.GameObj.Transform.Pos.Xy).LengthSquared >= movement.LengthSquared)
+	{
+		this.GameObj.Transform.Pos.MoveBy(movement);
+		yield return WaitUntil.NextFrame;
+	}
+	
+	this.GameObj.Transform.Pos.MoveTo(target);
+}
+```
+
+Let's see it more in detail:
+
+```csharp
+private IEnumerable<WaitUntil> MoveTo(Vector2 target)
+```
+
+this is our Coroutine definition; note that, being a method like all others, it can have parameters and it can have any kind of access modifier, as long as it is accessible by its caller.
+
+```csharp
+Vector2 movement = (target - this.GameObj.Transform.Pos.Xy).Normalized;
+```
+
+We don't need anymore to keep a private class variable; now we can just keep our movement data inside the Coroutine and declutter the rest of the class from things that are used only in specific cases
+
+```csharp
+while((target - this.GameObj.Transform.Pos.Xy).LengthSquared >= movement.LengthSquared)
+{
+	this.GameObj.Transform.Pos.MoveBy(movement);
+```
+
+This part should be pretty simple: just keep on moving our GameObject's position until it's close enough to the target location, so close actually that if we were to move one more time we would go past it. But why doesn't this piece of code loop 200 times and move the GameObject to the target location instantaneously? Because of this little line here
+
+```csharp
+yield return WaitUntil.NextFrame;
+```
+
+This is where the magic happens: this line is what allows the Coroutine to relinquish control to the main Update loop, and it is from where it will continue its execution the next frame, when it will be advanced until it encounters the next yield instruction. 
+This means that, being inside a loop, the code will run multiple times until the GameObject has finally moved close enough to its target location, and the loop will end.
+
+So, to recap, this is what actually happens with the Coroutine:
+
+```
+Coroutine is instantiated, calculates the value for `movement`, enters the loop, moves the GameObject a bit, and yields execution until the next frame
+an Update loop starts, the Coroutine resumes its execution, loops back, moves the GameObject a little more, and yields execution again
+an Update loop starts, the Coroutine resumes its execution, loops back, moves the GameObject a little more, and yields execution again
+...
+an Update loop starts, the Coroutine resumes its execution, loops back, but this time the GameObject is finally close enough to its target, so it exits the while, moves the GameObject to its final target position, and ends.
+```
```

Comment:
Maybe make this a list, not a code block, could be easier to read, especially on mobile / smaller screens. Not sure, just try resizing your browser window a lot an see how style / content adjusts.